src/SBarcodeScanner.h

```diff
@@ -26,6 +26,7 @@ class SBarcodeScanner : public QVideoSink
 
     Q_PROPERTY(QVideoSink* videoSink READ videoSink WRITE setVideoSink NOTIFY videoSinkChanged)
     Q_PROPERTY(QRectF captureRect READ captureRect WRITE setCaptureRect NOTIFY captureRectChanged)
+    Q_PROPERTY(bool cameraAvailable READ cameraAvailable)
```

Comment:
Missing NOTIFY signal. Also there is no setter defined, if this property has to be read-only from QML, you can define a setter but don't declare it in the Q_PROPERTY() macro. Then you can modify the property from C++ but not from QML